relay-server/src/utils/buffer.rs

```diff
@@ -67,3 +75,33 @@ impl BufferGuard {
         Ok(ManagedEnvelope::new(envelope, permit))
     }
 }
+
+/// Run the persistent buffer setup with migrations if the persistent envelope buffer is enabled.
+///
+/// This function internally creates a Tokio runtime, and executes all the configuration steps
+/// within its context. After the setup is done, used runtime will be dropped.
+pub fn setup_persistent_buffer(config: &Config) -> Result<(), project_buffer::BufferError> {
+    if !config.cache_persistent_buffer_enabled() {
+        return Ok(());
+    }
+
+    relay_log::info!("Configuring the persistent envelope buffer");
+
+    let options = SqliteConnectOptions::new()
+        .filename(PathBuf::from("sqlite://").join(config.cache_persistent_buffer_path()))
+        .journal_mode(SqliteJournalMode::Wal)
+        .create_if_missing(true);
```

Comment:
Why does this code exist 2x, once in `from_path` and once here?

Reply:
Right now I have no idea how to stop the Relay from an actor, I can produce the error, kill one actor but the rest of the actors are up and running. I want to make sure it fails on start if the Relay cannot read file or run the migrations.